core/src/main/java/com/datastrato/gravitino/connector/CatalogInfo.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.connector;
+
+import com.datastrato.gravitino.Audit;
+import com.datastrato.gravitino.Catalog;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.annotation.Evolving;
+import java.util.Map;
+
+/**
+ * CatalogInfo exposes all the information about a catalog to the connector interface.
+ *
+ * <p>This class object will be passed in through {@link CatalogOperations#initialize(Map,
+ * CatalogInfo)}, users can level this object to get the information about the catalog.
+ */
+@Evolving
+public final class CatalogInfo {
```

Comment:
It's best to add some documents about the difference between `CatalogEntity` and `CatalogInfo` here. besides, does the instance can only be converted from `CatalogEntity`?

Reply:
Yes, it's only from `CatalogEntity`. Using a new `CatalogInfo` instead of `CatalogEntity` is to expose as a public class to users, so users can use this without needing to expose `CatalogEntity`, which is used internally.